src/datasources/accounts/accounts.datasource.spec.ts

```diff
@@ -62,65 +63,87 @@ describe('AccountsDatasource tests', () => {
 
   describe('createAccount', () => {
     it('creates an account successfully', async () => {
-      const address = getAddress(faker.finance.ethereumAddress());
+      const createAccountDto = createAccountDtoBuilder().build();
 
       const result = await target.createAccount({
-        address,
+        createAccountDto,
         clientIp: faker.internet.ipv4(),
       });
 
       expect(result).toStrictEqual({
         id: expect.any(Number),
         group_id: null,
-        address,
+        address: createAccountDto.address,
+        name: createAccountDto.name,
         created_at: expect.any(Date),
         updated_at: expect.any(Date),
       });
 
       // check the account is stored in the cache
-      const cacheDir = new CacheDir(`account_${address}`, '');
+      const cacheDir = new CacheDir(`account_${createAccountDto.address}`, '');
       const cacheContent = await fakeCacheService.hGet(cacheDir);
       expect(JSON.parse(cacheContent as string)).toStrictEqual(
         expect.arrayContaining([
           expect.objectContaining({
             id: expect.any(Number),
             group_id: null,
-            address,
+            address: createAccountDto.address,
+            name: createAccountDto.name,
           }),
         ]),
       );
     });
 
     it('creates an account successfully if the clientIp is not a valid IP', async () => {
-      const address = getAddress(faker.finance.ethereumAddress());
+      const createAccountDto = createAccountDtoBuilder().build();
 
       const result = await target.createAccount({
-        address,
+        createAccountDto,
         clientIp: faker.string.sample(),
       });
 
       expect(result).toStrictEqual({
         id: expect.any(Number),
         group_id: null,
-        address,
+        address: createAccountDto.address,
+        name: createAccountDto.name,
         created_at: expect.any(Date),
         updated_at: expect.any(Date),
       });
 
       // check the account is stored in the cache
-      const cacheDir = new CacheDir(`account_${address}`, '');
+      const cacheDir = new CacheDir(`account_${createAccountDto.address}`, '');
       const cacheContent = await fakeCacheService.hGet(cacheDir);
       expect(JSON.parse(cacheContent as string)).toStrictEqual(
         expect.arrayContaining([
           expect.objectContaining({
             id: expect.any(Number),
             group_id: null,
-            address,
+            address: createAccountDto.address,
+            name: createAccountDto.name,
           }),
         ]),
       );
     });
 
+    it('should fail if the name already exists', async () => {
```

Comment:
Should we also check that the name hashes are unique?

Reply:
I totally agree, but I think we cannot test that at the datasource level as the hash is not included in the function input parameters, it is calculated internally. I've added a test to verify the hashes are unique (the uniqueness constraint is applied) to the migration tests (`00009_account-names.spec.ts`) in [30b3e87](https://github.com/safe-global/safe-client-gateway/pull/2081/commits/30b3e87bc8d1f8c303360f50992311594c200058)